presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowFilterPushDown.java

```diff
@@ -235,10 +235,10 @@ private static OptionalInt extractUpperBound(TupleDomain<Symbol> tupleDomain, Sy
                 upperBound--;
             }
 
-            if (upperBound > Integer.MAX_VALUE) {
-                return OptionalInt.empty();
+            if (upperBound >= 0 && upperBound <= Integer.MAX_VALUE) {
```

Comment:
If upperBound == 0, the optimizer will rewrite the expression to use TopNRowNumber, which doesn't allow the count to be 0.

Add a test for that scenario, too.